src/pcre2_compile.c

```diff
@@ -6169,21 +6176,163 @@ for (;; pptr++)
 
     if ((*pptr & CLASS_IS_ECLASS) != 0)
       {
+      eclass_op_info op_info;
+      PCRE2_SIZE previous_length = (lengthptr != NULL)? *lengthptr : 0;
+      int classones = 0;
+      int classzeros = 0;
+
       previous = code;
       *code++ = OP_ECLASS;
       code += LINK_SIZE;
-      if (!PRIV(compile_class_nested)(options, xoptions, &pptr, &code,
-                                      errorcodeptr, cb, lengthptr))
+      *code++ = 0;  /* Flags, currently zero. */
+      if (!PRIV(compile_class_nested)(options, xoptions, FALSE, &pptr, &code,
+                                      &op_info, errorcodeptr, cb, lengthptr))
         return 0;
-      PUT(previous, 1, (int)(code - previous));
 
-      zeroreqcu = reqcu;
-      zeroreqcuflags = reqcuflags;
-      if (firstcuflags == REQ_UNSET) firstcuflags = REQ_NONE;
-      zerofirstcu = firstcu;
-      zerofirstcuflags = firstcuflags;
+      if (lengthptr != NULL)
+        {
+        *lengthptr += code - previous;
+        code = previous;
+        /* (*lengthptr - previous_length) now holds the amount of buffer that
+        we require to make the call to compile_class_nested() with
+        lengthptr = NULL, and including the (1+LINK_SIZE+1) that we write out
+        before that call. */
+        }
 
-      break;   /* We are finished with this class */
+      /* Do some useful counting of what's in the bitmap. */
+      while (classones < 32 && op_info.bits[classones] == 0xff) classones++;
```

Comment:
If you use uint32_t type (can be a union), processing these is much faster.

Reply:
Hmm. There are existing uses of `uint8_t[8]`, such as `pcre2_real_code.start_bitmap` (which has a few 32-count loops).

But that makes sense, I'll replace with a `class_bits_storage`.